server/auth.py

```diff
@@ -1,67 +1,39 @@
-import binascii
-import hashlib
-import os
 from functools import wraps
 
+import requests
 from flask import jsonify, request
 
-from server import app, db
-from server.models import User
+from server.models import Account
 
 
-@app.route('/device/register', methods=['POST'])
-def register_user():
-    secret = os.environ.get('AUTH_SECRET')
-    auth_secret = request.form.get('auth_secret')
-    if auth_secret is None:
-        return jsonify({'error': 'Auth secret is not provided.'}), 400
-    if not auth_secret == secret:
-        return jsonify({'error': 'Auth secret is not correct.'}), 400
-
-    device_id = request.form.get('device_id')
-    try:
-        User.get_or_create(device_id=device_id)
-        return jsonify({'exists': True})
-    except ValueError as err:
-        return jsonify({'error': str(err)}), 400
-
-
-@app.route('/auth', methods=['GET'])
 def auth():
```

Comment:
I think you can remove `auth` and just rename `_auth` to `auth`, but I'm not sure

Reply:
You can but then you wouldn't be able to pass special arguments like I did for `nullable`. We may want to include others, such as `redirect`, in the future, which would redirect to Shibboleth instead of returning an error.